test/integration/GuardedStream.test.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  RepresentationMetadata,
+  TypedRepresentationConverter,
+  readableToString,
+  ChainedConverter,
+  guardedStreamFrom,
+  RdfToQuadConverter, BasicRepresentation, getLoggerFor,
+} from '../../src';
+import type { Representation,
+  RepresentationConverterArgs,
+  Logger } from '../../src';
+
+jest.mock('../../src/logging/LogUtil', (): any => {
+  const logger: Logger = { error: jest.fn(), debug: jest.fn(), warn: jest.fn(), info: jest.fn() } as any;
+  return { getLoggerFor: (): Logger => logger };
+});
+const logger: jest.Mocked<Logger> = getLoggerFor('GuardedStream') as any;
```

Comment:
But this is just copied from the unit test? Let's note that in #499 for when we fix.

Reply:
The behaviour being tested here is very similar, we just want to make sure the error is being logged (instead of being thrown as it used to be).

Reply:
That’s fine; it’s just about the repeated test code that we should clean up in #499.